src/main/scala/mesosphere/marathon/state/PathId.scala

```diff
@@ -69,14 +69,27 @@ case class PathId(path: List[String], absolute: Boolean = true) extends Ordered[
     path.zip(definition.path).forall { case (left, right) => left == right }
   }
 
-  override def toString: String = toString("/")
+  override def toString: String = {
+    lazy val str = toString("/")
```

Comment:
This will be executed every time `toString` is called.
Please use 
```
override val toString: String = toString("/")
```
this will be evaluated only once.